app/models/user.rb

```diff
@@ -1,4 +1,21 @@
 class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+  validates_inclusion_of :role, in: %w[admin author visitor subscriber]
+
+  def author?
+    role == 'author'
+  end
+
+  def visitor?
+    role == 'visitor'
+  end
+
+  def admin?
+    role == 'admin'
+  end
+
+  def subscriber?
+    role == 'subscriber'
+  end
```

Comment:
There's another way of dealing with roles that I showcased in my guide (using `enum`) that would NOT require creating a separate method for each role.

Reply:
We will look into that, thank you!